app/scripts/migrations/126.ts

```diff
@@ -0,0 +1,39 @@
+import { hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 126;
+
+/**
+ * This migration removes `providerConfig` from the network controller state.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(
+  state: Record<string, unknown>,
+): Record<string, unknown> {
+  if (
+    hasProperty(state, 'NetworkController') &&
+    isObject(state.NetworkController)
+  ) {
+    delete state.NetworkController.providerConfig;
```

Comment:
Currently, the function attempts to delete providerConfig without checking if it exists. To make the code more robust and avoid potential runtime errors, please add a check to verify that providerConfig exists on state.NetworkController before attempting to delete it. This can be done using `hasProperty` or by directly checking `state.NetworkController.providerConfig`.

Reply:
That isn't necessary.  `delete` no-ops if the field does not exist.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete

`If the property which you are trying to delete does not exist, delete will not have any effect and will return true.`